Consensys Diligence Audit Review

Sidu28 opened the pull request.

ChaoticWalrus commented on src/contracts/core/DelegationManager.sol:
```diff
@@ -50,7 +55,7 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
     {
         _initializePauser(_pauserRegistry, initialPausedStatus);
-        DOMAIN_SEPARATOR = keccak256(abi.encode(DOMAIN_TYPEHASH, bytes("EigenLayer"), block.chainid, address(this)));
+        DOMAIN_SEPARATOR = keccak256(abi.encode(DOMAIN_TYPEHASH, bytes("EigenLayer"), ORIGINAL_CHAIN_ID, address(this)));
```
think we should also account for 4.5.B here -- using `keccak256(bytes("EigenLayer"))` instead of `bytes("EigenLayer")` to better match EIP-712.
Should also perhaps discuss 4.5.C

Sidu28 replied:
addressed.

ChaoticWalrus commented on src/contracts/core/StrategyManager.sol:
```diff
@@ -150,7 +148,7 @@ contract StrategyManager is
         initializer
     {
-        DOMAIN_SEPARATOR = keccak256(abi.encode(DOMAIN_TYPEHASH, bytes("EigenLayer"), block.chainid, address(this)));
+        DOMAIN_SEPARATOR = keccak256(abi.encode(DOMAIN_TYPEHASH, bytes("EigenLayer"), ORIGINAL_CHAIN_ID, address(this)));
```
see comments in DelegationManager for matching logic in this contract

Sidu28 replied:
Addressed: [3d1b89c](https://github.com/Layr-Labs/eigenlayer-contracts/pull/14/commits/3d1b89c1c2e0627bf2e5679d6e7588119f23a829)

ChaoticWalrus commented on src/contracts/core/DelegationManager.sol:
```diff
@@ -97,7 +102,14 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
+
+        bytes32 digestHash;
+        if (block.chainid != ORIGINAL_CHAIN_ID) {
+            bytes32 domain_separator = keccak256(abi.encode(DOMAIN_TYPEHASH, bytes("EigenLayer"), block.chainid, address(this)));
```
4.5.B is again relevant here

Sidu28 replied:
Addressed: [3d1b89c](https://github.com/Layr-Labs/eigenlayer-contracts/pull/14/commits/3d1b89c1c2e0627bf2e5679d6e7588119f23a829)

ChaoticWalrus reviewed (changes requested):
Looking pretty good. Going to list a few minor things (in addition to my comments):

- We discussed adding appropriate comments/warnings for 4.1 & 4.2 -- not seeing those included in this PR
- 4.3 not addressed either
- we should discuss 4.4 -- I think we'll get persistent compiler warnings if we mark `underlyingToShares` and `sharesToUnderlying` as potentially state-modifying -- any thoughts on this?
- might want to add a note about intended use with 4.8 as well -- to discuss
- PR seems to be missing 4.11 -- should delete the unused `onlyEigenPod` modifier in the StrategyManager
- can you write up something on 4.12 (just a brief doc on the usage we have) and share it with me?

Sidu28 commented:
> Looking pretty good. Going to list a few minor things (in addition to my comments):
> 
> * We discussed adding appropriate comments/warnings for 4.1 & 4.2 -- not seeing those included in this PR
> * 4.3 not addressed either
> * we should discuss 4.4 -- I think we'll get persistent compiler warnings if we mark `underlyingToShares` and `sharesToUnderlying` as potentially state-modifying -- any thoughts on this?
> * might want to add a note about intended use with 4.8 as well -- to discuss
> * PR seems to be missing 4.11 -- should delete the unused `onlyEigenPod` modifier in the StrategyManager
> * can you write up something on 4.12 (just a brief doc on the usage we have) and share it with me?

- Now added: [9742f81](https://github.com/Layr-Labs/eigenlayer-contracts/pull/14/commits/9742f81f1e15f8b899c984ff90d96dd4c8c0820e) and [3765480](https://github.com/Layr-Labs/eigenlayer-contracts/pull/14/commits/37654806842b0d44a099161137cf4c81eb2b2df6)
- 4.3 addressed: [3765480](https://github.com/Layr-Labs/eigenlayer-contracts/pull/14/commits/37654806842b0d44a099161137cf4c81eb2b2df6)
- added warning for 4.8: [b675089](https://github.com/Layr-Labs/eigenlayer-contracts/pull/14/commits/b67508907092b6bfb4482604b6c0d46ab986b494)
- 4.11 was addressed in the first commit to this PR

ChaoticWalrus commented on src/contracts/pods/DelayedWithdrawalRouter.sol:
```diff
@@ -72,6 +72,9 @@ contract DelayedWithdrawalRouter is Initializable, OwnableUpgradeable, Reentranc
      * @notice Called in order to withdraw delayed withdrawals made to the `recipient` that have passed the `withdrawalDelayBlocks` period.
      * @param recipient The address to claim delayedWithdrawals for.
      * @param maxNumberOfDelayedWithdrawalsToClaim Used to limit the maximum number of delayedWithdrawals to loop through claiming.
+     * @dev 
+     *      WARNING: Note that the caller of this function cannot control where the funds are sent, but they can control when the 
+     *              funds are sent once the withdrawal becomes claimable.
```
hmm, OK

ChaoticWalrus commented:
PR is looking good now, but CI seems to be failing on a weird Foundry-related(?) error.
Let's sync on the final couple items offline

ChaoticWalrus commented:
We've agreed that leaving `StrategyBase.underlyingToShares` and `StrategyBase.sharesToUnderlying` marked as 'view' is OK.
Not using a version in the signature calculation also deemed acceptable.
I've just pushed some tiny changes to the StrategyWrapper contract to fix a few compiler warnings.
